src/test/java/org/opensearch/knn/index/AdvancedFilteringUseCasesIT.java

```diff
@@ -0,0 +1,449 @@
+     *                         "vector": [
+     *                             3
+     *                         ],
+     *                         "k": 10,
+     *                         "filter": {
+     *                             "nested": {
+     *                                 "path": "test_nested",
+     *                                 "query": {
+     *                                     "term": {
+     *                                         "test_nested.parking": "false"
+     *                                     }
+     *                                 }
+     *                             }
+     *                         }
+     *                     }
+     *                 }
+     *             }
+     *         }
+     *     }
+     * }
+     */
+    @SneakyThrows
+    public void testFiltering_whenNestedKNNAndFilterFieldWithNestedQueries_thenSuccess() {
+        for (final String engine : enginesToTest) {
+            // Set up the index with nested k-nn and metadata fields
+            createKnnIndex(INDEX_NAME, createNestedMappings(1, engine));
+            for (int i = 1; i <= NUM_DOCS; i++) {
+                // making sure that only 2 documents have valid filters
+                final String metadataFieldValue = i % 2 == 0 ? "false" : "true";
+                String doc = NestedKnnDocBuilder.create(FIELD_NAME_NESTED)
+                    .addVectors(FIELD_NAME_VECTOR, new Float[] { (float) i + 1 })
+                    .addVectorWithMetadata(FIELD_NAME_VECTOR, new Float[] { (float) i }, FIELD_NAME_METADATA, metadataFieldValue)
+                    .build();
+                addKnnDoc(INDEX_NAME, String.valueOf(i), doc);
+            }
+            forceMergeKnnIndex(INDEX_NAME);
+
+            // Build the query with both k-nn and filters as nested fields. The filter should also have a nested context
+            final XContentBuilder builder = XContentFactory.jsonBuilder().startObject().startObject(QUERY);
+            builder.startObject(TYPE_NESTED);
+            builder.field(PATH, FIELD_NAME_NESTED);
+            builder.startObject(QUERY).startObject(KNN).startObject(FIELD_NAME_NESTED + "." + FIELD_NAME_VECTOR);
+            builder.field(VECTOR, QUERY_VECTOR);
+            builder.field(K, k);
+            builder.startObject(FILTER_FIELD);
+            builder.startObject(TYPE_NESTED);
+            builder.field(PATH, FIELD_NAME_NESTED);
+            builder.startObject(QUERY);
+            builder.startObject(TERM_FIELD);
+            builder.field(FIELD_NAME_NESTED + "." + FIELD_NAME_METADATA, "false");
+            builder.endObject();
+            builder.endObject();
+            builder.endObject();
+            builder.endObject();
+            builder.endObject().endObject().endObject().endObject().endObject().endObject();
+
+            final String response = EntityUtils.toString(performSearch(INDEX_NAME, builder.toString()).getEntity());
```

Comment:
For searches, are we assuring ANN functionality gets hit as well as exact? i.e. we arent short circuiting to exact?

Reply:
Let me see if I can go around that, but then it will explode the number of tests. we can control the cases for Faiss but not for Lucene.

Reply:
For Lucene, Im not as concerned about. Just want to ensure that it works properly with Faiss

Reply:
sure I will add it for Faiss.

Reply:
Added the tests, to ensure that we are hitting the ANN search.